roles/online_docs/templates/mkdocs/docs/apptainer.md

```diff
@@ -0,0 +1,595 @@
+
+About Linux containers
+* are a method for running multiple isolated Linux systems (containers) on a control host using a single Linux kernel,
+* provides the cgroups functionality that allows limitation and prioritization of resources (CPU, memory, block I/O, network, etc.) without the need for starting any virtual machines.
+
+Apptainer is a container platform
+* highly optimized to be run on a laptop or high-performance cluster,
+* extremely fast and lightweight - as there is no virtualization, and images are simply executed just as any other program,
+* enables a secure way of running containers - images can be easily verified,
+* limits security risk, as it can only access user-accessible resources,
+* can directly run images from the Docker repository (via path `docker://...`).
+
+Apptainer container image
+* is a file (f.e. `ubuntu_18.04.sif`) containing all programs and libraries needed to execute that environment,
+* can contain a small subset of programs (with libraries), or entire Linux operating system (like Alpine, CentOS or Ubuntu),
+* is highly portable between various Linux operating systems and environments - it only requires installed Apptainer,
+* container filesystem environment is by default read-only when deployed (this can be partially changed by the user at the runtime - see section `Sandbox`),
+* is usually in format`.sif` (Singularity Image Format), `.sqfs` (SquashFS) or `.img` (image) ext3
+* can be also a directory instead of a single file, containing all the needed files, browsable by the user outside the container.
+
+
+## List of commonly used commands
+
+1. Pull container image from Docker repository
+
+    `$ apptainer pull docker://alpine:latest`
+
+2. Start an interactive shell within your container
+
+    `$ apptainer shell alpine_latest.sif`
+
+3. Run container (this executes its predefined runscript command)
+
+    `$ apptainer run alpine_latest.sif`
+
+4. Execute custom command inside the container
+
+    `$ apptainer exec alpine_latest.sif cat /etc/os-release`
+
+5. Execute a command directly from the Docker's repository container
+
+    `$ apptainer exec docker://busybox:latest busybox | head -n1`
+
+6. Build sandbox directory directly from the Docker image
+
+    `$ apptainer build --sandbox alpine_sandbox docker://busybox:latest`
+
+7. Shell into the sandbox, pretend to be a root user and permanently save all changes
+
+    `$ apptainer shell --fakeroot --writable alpine_sandbox`
+
+8. Convert sandbox to static image
+
+    `$ apptainer build alpine.sif alpine_sandbox`
+
+## Building Apptainer images
+
+In order to use apptainer containers, the image first needs to be built. Images can be built on top of another images (they can be local images or from the Docker repository), which can be done without elevated permissions. On the other hand, if an image is being built from scratch with f.e. `yum` installer, the elevated permissions are needed on the system to execute `sudo apptainer build ...`. 
```

Comment:
Correct me if I'm wrong, but AFAIK whether root permissions are required depends on the commands executed in the _recipe_ to build the image. If those commands (e.g. `yum` or `apt-get`) require root, then building the image requires root. Whether this is for a new image from scratch or for building on top of an existing image does not matter, right. When building an image from scratch, there might be a higher chance that you will need/want to use `yum` or `apt-get` to install additional packages, you can do that in an image build on top of another one too.

Reply:
Yes. And like I wrote: building on top of another image **usually** does not need elevated. Building from scratch **usually** does. All for the reasons you mentioned.

In these few sentences I reduced "usually" into a quasi-statements to simplify the explanation. Note **can be done** from building on top of another image. And a `are needed` for the installation from scratch, as that is the case when building fresh centos-on-centos, ubuntu-on-centos and due to the other issues also alpine-on-centos ===> therefore sentence: you need root.

Otherwise I can start naming exact examples on when I got an error and when not...